controllers/topology/internal/extensions/patches/variables/variables.go

```diff
@@ -19,80 +19,89 @@ package variables
 
 import (
 	"encoding/json"
-	"fmt"
 
 	"github.com/pkg/errors"
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/utils/pointer"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	"sigs.k8s.io/cluster-api/controllers/topology/internal/contract"
 )
 
 const (
-	// Builtin is the prefix of all builtin variables.
-	// NOTE: this prefix acts as a "namespace" preventing name collision between
-	// user defined variable and builtin variables.
-	Builtin = "builtin"
+	// BuiltinVariableName is the name of the builtin variable.
+	// NOTE: User-defined variables are not allowed to start with "builtin".
+	BuiltinVariableName = "builtin"
```

Comment:
```suggestion
	// BuiltinsMapKey is the name of the builtin variable.
	// NOTE: User-defined variables are not allowed to start with "builtin".
	BuiltinsMapKey = "builtin"
```

Reply:
I went with `BuiltinsName` for now. Are you sure you want to call it `BuiltinsMapKey` because (from the users point of view) it is actually the name of the builtins variable?